docs/core/idioms/secrets.md

```diff
@@ -0,0 +1,102 @@
+# Authentication and Secrets in Prefect
```

Comment:
Let's not expose this one yet; we can keep the text in the repo but whenever 0.11.0 goes out this will change slightly